msr-flows/msbr_compare/run_user_input.py

```diff
@@ -341,159 +433,378 @@ def cycle_time_decay(self, identifier='CTD'):
 
 if __name__ == '__main__':
 
-    output_path = f'./{ui.path_to_dump_files}/'
-    misc_funcs.set_directory(output_path)
-    close_boolean = False
+    base_output_path = f'./{ui.path_to_dump_files}/'
+    misc_funcs.set_directory(base_output_path)
+
     element_dictionary = dict()
+    active_identifiers = list()
 
     for index in range(len(ui.element_flow_list)):
         element_dictionary[ui.element_flow_list[index]] = [
             ui.associated_symbol_list[index], ui.associated_atomic_list[index]]
+    for N_index, N_steps in enumerate(ui.number_serp_steps_list):
+        output_path = str(base_output_path) + f'{N_steps}/'
+        misc_funcs.set_directory(output_path)
+        print(f'Step size: {(ui.end_time - ui.start_time) / N_steps}')
+
+        if ui.control:
```

Comment:
Sections 450 -473, 475-502, 504-533, 535-558, 560-583, 585-608 could be one function that calls different methods. Avoid unnecessary code repetitions.